### PR TITLE
Perf-tests golint checks fix

### PR DESCRIPTION
**Pull request by qckzr**
Fixed golint checks and removed -min_confidence flag from verify-golint.sh as mentioned in #1163

 File  `clusterloader2/pkg/state/state.go` had the following warnings:
 

>exported method GetNamespacesState returns unexported type *state.namespacesState, which can be annoying to use
exported method GetResourcesVersionState returns unexported type *state.resourcesVersionsState, which can be annoying to use 

So I exposed both NamespacesState and ResourcesVersionsState types;

I already ran the verify scripts locally without issues, this is my first time contributing in this project so apologies if I made a mistake

### CODE REVIEW

File: clusterloader2/pkg/state/namespaces_state.go [hunk 1]
```diff
@@ -34,28 +34,28 @@ type InstancesState struct {
...
+	APIGroup   string
 }
 
 // namespaceState represents state of a single namespace.
 type namespaceState map[InstancesIdentifier]*InstancesState
 
-// namespacesState represents state of all used namespaces.
-type namespacesState struct {
+// NamespacesState represents state of all used namespaces.
+type NamespacesState struct {
```
Comment (wojtek-t) [inline clusterloader2/pkg/state/namespaces_state.go:44]: please don't expose it - it wasn't exposed before

File: clusterloader2/pkg/state/resource_state.go [hunk 2]
```diff
@@ -25,26 +25,26 @@ import (
...
-	ApiGroup   string
+	APIGroup   string
 }
 
-// resourcesVersionsState represents most recent resources versions for a given object types.
+// ResourcesVersionsState represents most recent resources versions for a given object types.
 // These versions are available only for resource types of objects that were create/updated
 // by the clusterloader.
-type resourcesVersionsState struct {
+type ResourcesVersionsState struct {
```
Comment (wojtek-t) [inline clusterloader2/pkg/state/resource_state.go:34]: don't expose - it wasn't exposed before

File: clusterloader2/pkg/state/namespaces_state.go (hunk 1 above)
Comment (qckzr) [reply clusterloader2/pkg/state/namespaces_state.go:44]: My concern is if I don't expose them, we'll get error when running `verify-golint.sh` (unless flag `-min_confidence=0.9` is used).

I tried to look if it's possible to disable golint on file `clusterloader2/pkg/state/state.go` but looks like that's not possible,

Thoughts?

Comment (wojtek-t) [reply clusterloader2/pkg/state/namespaces_state.go:44]: OK - I missed the PR description. It makes sense for now - we can try figuring out if we can somehow hide it later.

**Comment by wojtek-t** [comment]
/approve

@qckzr - please rebase your PR and I will lgtm.

File: clusterloader2/pkg/prometheus/prometheus.go [hunk 3]
```diff
@@ -185,10 +185,9 @@ func (pc *PrometheusController) SetUpPrometheusStack() error {
 }
 
 // TearDownPrometheusStack tears down prometheus stack, releasing all prometheus resources.
-func (pc *PrometheusController) TearDownPrometheusStack() error {
-	// Get disk metadata again to be sure
-	if err := pc.cachePrometheusDiskMetadataIfEnabled(); err != nil {
-		klog.Warningf("Error while caching prometheus disk metadata: %v", err)
+func (pc *Controller) TearDownPrometheusStack() error {
+	if err := pc.snapshotPrometheusDiskIfEnabled(); err != nil {
```
Comment (wojtek-t) [inline clusterloader2/pkg/prometheus/prometheus.go:189]: This isn't a good rebase - you changed things.

File: network/benchmarks/netperf/nptest/nptest.go [hunk 4]
```diff
@@ -158,24 +157,19 @@ func init() {
...
-		{SourceNode: "netperf-w1", DestinationNode: "netperf-w3", Label: "13 iperf UDP. Remote VM using Pod IP", Type: iperfUdpTest, ClusterIP: false, MSS: mssMax},
-		{SourceNode: "netperf-w3", DestinationNode: "netperf-w2", Label: "14 iperf UDP. Remote VM using Virtual IP", Type: iperfUdpTest, ClusterIP: true, MSS: mssMax},
-		{SourceNode: "netperf-w1", DestinationNode: "netperf-w2", Label: "15 netperf. Same VM using Pod IP", Type: netperfTest, ClusterIP: false},
-		{SourceNode: "netperf-w1", DestinationNode: "netperf-w2", Label: "16 netperf. Same VM using Virtual IP", Type: netperfTest, ClusterIP: true},
-		{SourceNode: "netperf-w1", DestinationNode: "netperf-w3", Label: "17 netperf. Remote VM using Pod IP", Type: netperfTest, ClusterIP: false},
-		{SourceNode: "netperf-w3", DestinationNode: "netperf-w2", Label: "18 netperf. Remote VM using Virtual IP", Type: netperfTest, ClusterIP: true},
+		{SourceNode: "netperf-w1", DestinationNode: "netperf-w2", Label: "1 iperf TCP. Same VM using Pod IP", Type: iperfTCPTest, ClusterIP: false, MSS: mssMin},
```
Comment (wojtek-t) [inline network/benchmarks/netperf/nptest/nptest.go:160]: Same here - this doesn't seem to be a good rebase, some cases are missing.

File: network/benchmarks/netperf/nptest/nptest.go [hunk 5]
```diff
@@ -412,16 +406,7 @@ func parseIperfTcpBandwidth(output string) string {
...
-	// Parses the output of iperf3 and grabs the group Mbits/sec from the output
-	match := iperfSCTPOutputRegexp.FindStringSubmatch(output)
-	if match != nil && len(match) > 1 {
-		return match[1]
-	}
-	return "0"
-}
-
-func parseIperfUdpBandwidth(output string) string {
+func parseIperfUDPBandwidth(output string) string {
```
Comment (wojtek-t) [inline network/benchmarks/netperf/nptest/nptest.go:409]: Same here

File: network/benchmarks/netperf/nptest/nptest.go [hunk 6]
```diff
@@ -461,30 +446,21 @@ func (t *NetPerfRpc) ReceiveOutput(data *WorkerOutput, reply *int) error {
...
-
-	case iperfSctpTest:
-		mss := testcases[currentJobIndex].MSS - mssStepSize
-		outputLog = outputLog + fmt.Sprintln("Received SCTP output from worker", data.Worker, "for test", testcase.Label,
-			"from", testcase.SourceNode, "to", testcase.DestinationNode, "MSS:", mss) + data.Output
-		writeOutputFile(outputCaptureFile, outputLog)
-		bw = parseIperfSctpBandwidth(data.Output)
-		cpuSender, cpuReceiver = parseIperfCpuUsage(data.Output)
+		bw = parseIperfTCPBandwidth(data.Output)
+		cpuSender, cpuReceiver = parseIperfCPUUsage(data.Output)
```
Comment (wojtek-t) [inline network/benchmarks/netperf/nptest/nptest.go:455]: and here...

Please fix everywhere.

File: network/benchmarks/netperf/nptest/nptest.go (hunk 4 above)
Comment (qckzr) [reply network/benchmarks/netperf/nptest/nptest.go:161]: Yeah, apologies for the issue, I made a mistake while rebasing the repo, but now it should be all good..!

**Pull request merged**